documentation/directory-structure.md

```diff
@@ -0,0 +1,201 @@
+  - [Resource provider namespace (`rpnamespace`) folders layout](#resource-provider-namespace-rpnamespace-folders-layout)
+  - [API version (`api_version`) folders layout](#api-version-api_version-folders-layout)
+    - [`examples` folder](#examples-folder)
+  - [Naming guidelines for `specification` folder contents](#naming-guidelines-for-specification-folder-contents)
+  - [Example layout of the `specification` folder](#example-layout-of-the-specification-folder)
+  - [Folder Structure for service groups](#folder-structure-for-service-groups)
+  - [common-types](#common-types)
+  - [FAQ](#faq)
+    - [Should my service team have one or multiple resource provider folders?](#should-my-service-team-have-one-or-multiple-resource-provider-folders)
+
+This article described the directory structure / folder layout of the `specification` folder.
+
+The layout described in this article is strictly enforced. There exist
+some exceptions for historical reasons. These exceptions are not allowed going forward.
+
+## `specification` folder
+
+The `specification` folder is the root folder for all TypeSpec and OpenAPI specs 
+and related documents. Each child of `specification` corresponds to a `resource provider`.
+There is a special case of a `service group`: see [Folder Structure for service groups](#folder-structure-for-service-groups).
+
+## Resource provider (`rp`) folder
+
+Each child folder of the `specification` folder is a root of given `resource provider`.
+
+Example resource providers root folders:
+
+- [`specification/containerservice`]
+- [`specification/confidentialledger`]
+
+Given resource provider `rp` has following structure,
+for multiple values of `rpnamespace` (`rpnamespace` stands for **resource provider namespace**; see [relevant section](#resource-provider-namespace-rpnamespace-folders)):
+
+- `specification/rp/rpnamespace`
+- `specification/rp/data-plane/rpnamespace`
+- `specification/rp/resource-manager/rpnamespace`
+
+For a list of existing resource providers and their corresponding Azure services,
+consult the [resource provider list].
+
+### TypeSpec sources
+
+The `specification/rp/rpnamespace` directories contain TypeSpec sources.  
+To learn about TypeSpec, see [aka.ms/azsdk/typespec] and [aka.ms/typespec].  
+To learn these directories structure, see [TypeSpec directory structure] article.  
+The rest of this article pertains to the `specification` directory structure 
+**excluding** the contents of `specification/rp/rpnamespace`.
+
+## `resource-manager` and `data-plane` folders
+
+The OpenAPI specs in `specification/rp/resource-manager` pertain to ARM resources while
+the OpenAPI specs in `specification/rp/data-plane` pertain to everything else.
+
+There should be an AutoRest configuration file (`readme.md`) at the root of each
+of these directories when present.
+
+If TypeSpec sources are present, OpenAPI specs in the `resource-manager`
```

Comment:
I got a little confused here. Maybe word it that if in the root rp folder there is a TypeSpec file, the OpenAPI specs created from that TypeSpec are located here...